src/rust/engine/process_execution/src/lib.rs

```diff
@@ -454,6 +454,20 @@ pub enum ProcessExecutionStrategy {
   Docker(String),
 }
 
+impl ProcessExecutionStrategy {
+  /// What to insert into the Command proto so that we don't incorrectly cache
+  /// Docker vs remote execution vs local execution.
+  pub fn cache_value(&self) -> String {
+    match self {
+      Self::Local => "local_execution".to_string(),
+      Self::RemoteExecution(_) => "remote_execution".to_string(),
```

Comment:
@illicitonion thoughts if this should include something like the execution address? We already include elsewhere the platform properties. But it's possible to use the same platform properties with two different execution servers, which could result in reusing the same cache when it's illegal.

I suspect it's a non-issue because this is not common to happen, and there's a workaround by changing `process_execution_cache_namespace`.